NPRML Construction Changes

jwcounts opened the pull request:
These modifications made to classes/nprml.php help with sending media file attachments to the NPR API/NPR One from Wordpress sites that use media offload plugins, such as WP Offload Media by Delicious Brains.  The full scope of the modifications is detailed in [this issue on the base fork](https://github.com/npr/nprapi-wordpress/issues/64).

It is currently working on [our production site](https://www.houstonpublicmedia.org), as well as [KTOO's website](https://ktoo.org). It has also been tested on a base Wordpress install and one with the Powerpress podcasting plugin installed.

Commit:
NPRML Construction Changes

NPRML class uses wp_get_attachment_url() instead of attachment GUID
Deduplication between direct attachments and enclosures

davidmpurdy commented:
I think it's worth noting that this isn't just about support for one specific plugin. I believe it also lines things up more generally with Wordpress best practices (e.g., not relying on GUID to be a link to the media file) that could help in other situations as well (such as when a site has migrated to another domain and GUIDs have properly been left unchanged).

tamw-wnet committed:
resolves error raised when any non-admin user tries to trash a post

tamw-wnet committed:
Merge pull request #3 from tamw-wnet/fix_trash_post_error

resolves error raised when any non-admin user tries to trash a post

Commit:
Change image_guid variable name to match content

jwcounts committed:
NPRML Construction Changes (#1)

* NPRML Construction Changes

NPRML class uses wp_get_attachment_url() instead of attachment GUID
Deduplication between direct attachments and enclosures

* Change image_guid variable name to match content

Co-authored-by: Jared Counts <jcounts@houstonpublicmedia.org>

jwcounts committed:
Bug fix for query when activating in multisite

jwcounts committed:
Took a sweep for consistent styling:
- arrays use square brackets instead of keyword
- Indents are tabs equal to 4 spaces
- Conditionals use {} instead of keywords (endif, etc)
- spaces within and around () unless they're empty
- removed spaces after ! in conditionals

*NOTE*: none of the code actually changed
This was strictly for my own sanity/edification

tamw-wnet committed:
Generate WP posts with rich layout including multimedia etc (#4)

* If 'layout' is available in the NPR Story API output, generate complex HTML to render any images, YouTube videos, Tweets, iframes, or JavaScript-based widgets within the post in the order they appeared on the NPR website.

* if using the NPR "layout" only sideload primary story image into WordPress Media Library instead of all referenced images in the story

* Adding help text for layout option

Updating the text for the layout checkbox explaining that it may require a ticket with NPR to update their Story API key to get the layout section.

* Further tweaks to the help text for the

bdivver committed:
Select a default category for each query when using post (#7)

* Update settings.php

* Update NPRAPIWordpress.php

* Update NPRAPIWordpress.php

Co-authored-by: William Tam (WNET) <tamw-wnet@users.noreply.github.com>

jwcounts committed:
Styling changes
Bug fixes
Resolving merge conflicts with main branch

jwcounts committed:
Cleaned up duplicate variable

jwcounts committed:
Merge branch 'master' into master
